sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientCertificateCredentialBuilder.java

```diff
@@ -86,9 +116,14 @@ public ClientCertificateCredential build() {
         ValidationUtil.validate(getClass().getSimpleName(), new HashMap<String, Object>() {{
                 put("clientId", clientId);
                 put("tenantId", tenantId);
-                put("clientCertificate", clientCertificate);
+                put("clientCertificate", clientCertificate == null ? clientCertificatePath : clientCertificate);
             }});
-        return new ClientCertificateCredential(tenantId, clientId, clientCertificate, clientCertificatePassword,
-            identityClientOptions);
+        if (clientCertificate != null && clientCertificatePath != null) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("Both certificate input stream and "
+                    + "certificate path are provided in ClientCertificateCredentialBuilder. Only one of them should "
+                    + "be provided."));
+        }
```

Comment:
If we are doing this, then maybe the `ClientCertificateCredential` constructor should only have the input stream arg. If the path is given, the builder can wrap the path with `FileInputStream` and create the credential instance. This saves multiple `if-else` checks in `ClientCertificateCredential`.